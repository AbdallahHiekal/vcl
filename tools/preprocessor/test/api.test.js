'use strict';

// var assert = require('assert');
var vcl = require('../index');

var opts = {root: __dirname};

describe('vcl', () => {
  it('should be a function', () => {
    expect(typeof vcl).toBe('function');
  });

  it('should throw if there is no css input', async () => {
    return expect(vcl()).rejects.toBeInstanceOf(Error);
  });
});

describe('vcl object', function() {

  it('should return valid css', async function() {
    var result = await vcl('body\n  color: blue');
    expect(typeof result.css).toBe('string');
    expect(result.css).toMatchSnapshot();

  });

  it('should return valid css without any options set', async function() {
    let result = await vcl('body\n  color: rgba(#ccc, .5)');
    expect(result.css).toMatchSnapshot();
  });

  it('should include local css files', async function() {
    const result = await vcl(`@import "./fixtures/test-import.css"\na\n  color: green`, opts);
    expect(typeof result.css).toBe('string');
    expect(result.css).toMatchSnapshot();
  });

  it('should include local sss files', async function() {
    const result = await vcl(`@import "./fixtures/test-import.css"\na\n  color: green`, opts);
    expect(typeof result.css).toBe('string');
    expect(result.css).toMatchSnapshot();
  });

  it('should include npm imports', async function() {
    const result = await vcl('@import "@vcl/theme"\na\n  color: green', opts);
    expect(typeof result.css).toBe('string');
    expect(result.css.length).toBeGreaterThan(100);
  });


  it('should not remove the comment', async function() {
    const result = await vcl('/* will not be removed */\nbody\n  color: blue\n', opts);
    expect(typeof result.css).toBe('string');
    expect(result.css).toMatchSnapshot();
  });

  it('should remove the comment', async function() {
    const result = await vcl(
`/* will be removed */
body
  color: blue
`, {
      ...opts,
      optimize: true
    });
    expect(typeof result.css).toBe('string');
    expect(result.css).toMatchSnapshot();
  });

  it('should override the theme var', async function() {
    const result = await vcl(
`@import "@vcl/theme"
body
  color: blue
@import "./fixtures/custom-theme.sss"
`, {
      ...opts,
      optimize: true
    });

    expect(typeof result.css).toBe('string');
    expect(result.css).toMatchSnapshot();
  });

  it('should contain source maps', async function() {
    var result = await vcl('body\n  color: blue', {
      ...opts,
      sourceMap: true
    });
    expect(result.css).toMatchSnapshot();
  });

});
