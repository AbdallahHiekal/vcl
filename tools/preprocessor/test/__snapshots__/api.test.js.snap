// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vcl object should contain source maps 1`] = `
"body {
  color: blue
}
/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInZjbC5zc3MiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUE7RUFDRTtBQURDIiwiZmlsZSI6InZjbC5jc3MiLCJzb3VyY2VzQ29udGVudCI6WyJib2R5XG4gIGNvbG9yOiBibHVlIl19 */"
`;

exports[`vcl object should include local css files 1`] = `
"body {
  background: red;
}
a {
  color: green;
}"
`;

exports[`vcl object should include local sss files 1`] = `
"body {
  background: red;
}
a {
  color: green;
}"
`;

exports[`vcl object should not remove the comment 1`] = `
"/* will not be removed */
body {
  color: blue
}
"
`;

exports[`vcl object should override the theme var 1`] = `".body{background-color:#d9534f}body{color:#00f}"`;

exports[`vcl object should remove the comment 1`] = `"body{color:#00f}"`;

exports[`vcl object should return valid css 1`] = `
"body {
  color: blue
}"
`;

exports[`vcl object should return valid css without any options set 1`] = `
"body {
  color: rgba(204,204,204, .5)
}"
`;
